File: src/Rotslyn.Transpilers.Tests/Samples/BasicClassWithMixedProperties.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Rotslyn.Transpilers.Tests.Samples
{
    public class BasicClassWithMixedProperties
    {
        private int NormalPrivateProperty { get; set; }
        protected int NormalProtectedProperty { get; set; }
        public int NormalPublicProperty { get; set; }

        int PrivatePropertyWithNoAccessors { get; set; }
        public int PublicPropertyWithPrivateGetAccessorAndNormalSetAccessor { private get; set; }
        public int PublicPropertyWithProtectedGetAccessorAndNormalSetAccessor { protected get; set; }
        protected int ProtectedPropertyWithPrivateGetAccessorAndNormalSetAccessor { private get; set; }

        public int PublicPropertyWithPrivateGetterOnly { get; }
        private int PrivatePropertyWithProtectedGetterOnly { get; }
        protected int ProtectedPropertyWithPrivateGetterOnly { get; }
        
        public int PublicPropertyWithNoGetAccessorAndProtectedSetAccessor { get; protected set; }
        public int PublicPropertyWithNoGetAccessorAndPrivateSetAccessor { get; private set; }


        private int PrivateMember;
        protected int ProtectedMember;
        public int PublicMember;
        int MemberWithNoAccessors;
    }
}

File: src/Rotslyn.Transpilers.Tests/Samples/BasicClassWithMixedProperties.ts
module Rotslyn.Transpilers.Tests.Samples {
    export class BasicClassWithMixedProperties {

        private normalPrivateProperty: number;

        protected normalProtectedProperty: number;

        public normalPublicProperty: number;

        private privatePropertyWithNoAccessors: number;

        public publicPropertyWithPrivateGetAccessorAndNormalSetAccessor: number;

        public publicPropertyWithProtectedGetAccessorAndNormalSetAccessor: number;

        public protectedPropertyWithPrivateGetAccessorAndNormalSetAccessor: number;

        private _____publicPropertyWithPrivateGetterOnly: number;
        public get publicPropertyWithPrivateGetterOnly(): number {
            return this._____publicPropertyWithPrivateGetterOnly;
        }

        private _____privatePropertyWithProtectedGetterOnly: number;
        private get privatePropertyWithProtectedGetterOnly(): number {
            return this._____privatePropertyWithProtectedGetterOnly;
        }

        protected _____protectedPropertyWithPrivateGetterOnly: number;
        protected get protectedPropertyWithPrivateGetterOnly(): number {
            return this._____protectedPropertyWithPrivateGetterOnly;
        }

        private privateMember: number;

        protected protectedMember: number;

        public publicMember: number;

        private memberWithNoAccessors: number;
    }
}